Removed br tags

tanishq-arya opened the pull request:
# Related Issue
Fixes: #231 

# Prosposed Changes
-Removed <br> tags
- Added margin and padding to the section element.
- Added a section around initiatives 

# Additional Info
- any additional information or context

# Screenshots

Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

garg3133 commented on templates/new_home/index.html:
```diff
@@ -134,30 +133,31 @@ <h2>MISSION</h2>
+                <img style="height: 300px;" alt="image"
+                    src="{% if initiative.thumbnail %}{{ initiative.thumbnail.url }}{% else %}{{ initiative.thumb_url }}{% endif %}">
+            </div>
+            <br>
```
You skipped this `br`.

garg3133 commented on static/new_home/style.css:
```diff
@@ -12,6 +12,12 @@ body
 }
 
+section {
+  box-sizing: border-box;
```
This property is already applicable everywhere with Bootstrap. You don't need to explicitly add it.

garg3133 commented on static/new_home/style.css:
```diff
@@ -12,6 +12,12 @@ body
 
+section {
+  box-sizing: border-box;
+  padding: 3em 0;
```
This adds a lot of unwanted space at the top of every section. Please fix it.

garg3133 commented on templates/new_home/index.html:
```diff
@@ -134,30 +133,31 @@ <h2>MISSION</h2>
+    <div>
+        <center>
+            <h1 style="padding-top: 20px;" data-aos='fade-up'> Inititatives</h1>
+        </center>
```
Instead of `center` tag, a `text-center` class can be added in the parent `div`.

tanishq-arya replied:
This was already there I just made a section around this.

garg3133 replied:
Yeah, I know that. But it'll be better if it also gets fixed here only.

garg3133 reviewed (commented):
@tanishq-arya All the changes you made in the CSS files seems unnecessary. You can just add some Bootstrap margins classes to each section to achieve the required result.

tanishq-arya commented:
Working on the changes. But I think this CSS is more useful than bootstrap as some classes might cause the view to change like the extra spaces.

garg3133 commented:
> Working on the changes. But I think this CSS is more useful than bootstrap as some classes might cause the view to change like the extra spaces.

You can always achieve the required spacing by using some combination of margins and paddings. Anyways, there's no hard and fast rule of keeping the spacing at the current values only (which are quite inconsistent), instead using some consistent margins for all the sections will be better.

Also, the main motive of using CSS frameworks like bootstrap is that we need to write as less custom CSS as possible, so if it can be done using Bootstrap, it should be done using Bootstrap.

tanishq-arya commented:
I removed the excess css and added 2 classes to donate us section for margin and padding. If there are any other changes required please let me know.

garg3133 commented on static/new_home/style.css:
```diff
@@ -84,6 +84,7 @@ nav
   margin: 10px;
   color: #fff;
   background: #eb5d1e;
+  cursor: pointer;
```
It was causing problem because the button was wrapped being wrapped by a `p` tag instead of an `a` tag. I have fixed this in the latest commit, so you can remove this from here.

garg3133 reviewed (commented):
@tanishq-arya You've removed most of the `br` tags, but haven't replaced them with appropriate margins due to which the structure of some sections is breaking. Kindly add appropriate margins (using bootstrap's margin classes) to these sections.

One more thing to take care is that the links in the navbar (About and Contact) should take the user  at the appropriate position. Currently, the heading of the section gets covered by navbar on clicking those links.

tanishq-arya commented:
Will push requested changes soon.

tanishq-arya commented:
> @tanishq-arya You've removed most of the `br` tags, but haven't replaced them with appropriate margins due to which the structure of some sections is breaking. Kindly add appropriate margins (using bootstrap's margin classes) to these sections.
> 
> One more thing to take care is that the links in the navbar (About and Contact) should take the user at the appropriate position. Currently, the heading of the section gets covered by navbar on clicking those links.

I found a solution for the overlapping it was implemented already using the ::before for about us and contact us I increased the size and margin to 65px which is more than nav 63px.

garg3133 commented on static/new_home/style.css:
```diff
@@ -582,7 +581,7 @@ header
   content: "";
-  /* margin-top: -25px; */
+  margin-top: -25px;
   height: 25px;
```
I think you should change to `65px` here also.

tanishq-arya replied:
> I have my end-sem examinations coming up in a few days and the due dates for all my assignments and projects are also approaching, so I may not be active in reviewing PRs for a few days.

No issues. All the best.

garg3133 reviewed (commented):
I have my end-sem examinations coming up in a few days and the due dates for all my assignments and projects are also approaching, so I may not be active in reviewing PRs for a few days.

tanishq-arya commented:
Also I was thinking in mobile view the contact us forms should have some spacing between them should I make a new issue for that ?

garg3133 reviewed (commented):
The spacing of different sections is very inconsistent. Like About Us section has too much space above but very less space (padding) at the bottom of the section. Initiative section hardly has any space at the bottom and then the Donate Us section has too much space at the top.

Also the animations are not working properly on small screen (see animation of Welcome to Jagrati part for instance on a mobile phone, it does not appear in the first frame).

tanishq-arya commented:
> The spacing of different sections is very inconsistent. Like About Us section has too much space above but very less space (padding) at the bottom of the section. Initiative section hardly has any space at the bottom and then the Donate Us section has too much space at the top.
> 
> Also the animations are not working properly on small screen (see animation of Welcome to Jagrati part for instance on a mobile phone, it does not appear in the first frame).

I think there were changes after this too. I'll update the PR.

tanishq-arya commented:
@garg3133 I found what happened. Earlier the nav was fixed on top so everything was hidden below it. Now the nav collapses so space is left. I'll fix and update the PR.

tanishq-arya commented:
@garg3133 Can you check now? And I didn't understand about the animation. I didn't change anything related to the animations.

garg3133 reviewed (commented):
@tanishq-arya Everything is still very inconsistent:
* On clicking the links in the navbar, the title of that section gets hidden.
* You've used `my-3 py-4` classes at some places and just `my-3` at others. The classes being used for every section should be same.
* I see that will cause some inconsistency in spacing, as at some places (ex. About Us section), there is no top padding in the title while in Initiatives section, there is a top padding in the title. So the correct way to go from here would be to remove the top-paddings from all the titles.
* In the Donate Us section, `br` tag is being used to add space below the title. Instead, margin/padding should be used. Same for the Contact Us section. You can use `mb-3` margin below the title instead, if that looks fine.
* The card-container of Initiatives section adds too much padding at the top and bottom. So maybe try to reduce that a bit, so that the padding seems consistent at the bottom if the section?

tanishq-arya commented:
I did that because there was different content in each. 

So now I should make all the sections my-3 py-4 ?

I'll remove the donate section <br>'s too.

tanishq-arya commented:
> I did that because there was different content in each.
> 
> So now I should make all the sections my-3 py-4 ?
> 
> I'll remove the donate section
> 's too.

@garg3133 any update on this ? What should I do ?

garg3133 commented:
> So now I should make all the sections my-3 py-4 ?

I think just the padding should be fine in most cases. You can add the margins if required.

tanishq-arya commented:
@garg3133 I have added my-3, py-4 to all sections. Also I have added mx-auto to keep every section aligned in center.

tanishq-arya commented:
Can you take a look at PR #277 it was a small padding fix, and the forms look much better. I have added the screenshots too.
